File: Lab-4/TOH.py
import time

def TOH(n, src, dst, temp):
    if n==1:
        print(f"Move disk {n} from {src} to {dst}")
    else:
        TOH(n-1, src, temp, dst)
        print(f"Move disk {n} from {src} to {dst}")
        TOH(n-1, temp, dst, src)

n = int(input("Enter number of disks:"))

start_time = time.time()
TOH(n, 'A', 'C','B')
end_time = time.time()

print(f"Time taken: {end_time - start_time:.6f} seconds")

File: Lab-4/TOH.c
#include<stdio.h>
#include<time.h>
long int TOH(int n, char src, char dst, char temp)
{
    if(n==1)
    {
        printf("Move disk %d from %c to %c\n ",n,src,dst);
    }
    else{
        TOH(n-1,src,temp,dst);
        printf("Move disc %d from %c to %c\n",n,src,dst);
        TOH(n-1,temp,dst,src);
    }
}
int main()
{
    int n;
    time_t start,end;
    double diff;
    printf("Enter the value of n:");
    scanf("%d",&n);
    start=time(NULL);
    TOH(n,'A','C','B');
    end=time(NULL);
    diff=difftime(end,start);
    printf("The time taken is %lf seconds\n",diff);
    return 0;

}